base/intfuncs.jl

```diff
@@ -369,6 +369,7 @@ julia> powermod(5, 3, 19)
 ```
 """
 function powermod(x::Integer, p::Integer, m::T) where T<:Integer
+    p < 0 && p == typemin(typeof(p)) && return mod(powermod(invmod(x, m), -(p÷2), m)^2, m)
```

Comment:
Although this for most sensible integer types it would give wrong results if `typemin(typeof(p))` is odd so it would probably be good to check for that. Also, I believe this will cause `powermod(2, big(-3), 2)` to fail since `BigInt` doesn't have `typemin` defined. The better check would probably be `p==-p` (with a comment about how this is true for typemin)

Reply:
Thank you for the suggestions! I have optimized it, but I wonder is it necessary to check the typemin(typeof(p)) (in this case, it is equals to p) is even or odd. Because for the signed integers in the form of 2's complement repretation, the minimum will always be even

Reply:
it  would probably not matter in practice, but the iseven(typemin(T)) should constant fold making it free, and it will prevent a really weird bug when someone calls this with a very odd integer type.

Reply:
got it